### PR TITLE
CVP-3853: Add HyperShift workflow to existing v4.12 job & config

### PR DESCRIPTION
**Pull request by gastoner**
Please review @asergienk

### CODE REVIEW

**Comment by asergienk** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by asergienk** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.13-cvp-common-claim

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by asergienk** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by asergienk** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by asergienk** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

File: ci-operator/config/redhat-openshift-ecosystem/cvp/redhat-openshift-ecosystem-cvp-ocp-4.12.yaml [hunk 1]
```diff
@@ -35,6 +35,13 @@ tests:
       PYXIS_URL: ""
       TEST_MODE: cvp
     workflow: optional-operators-cvp-common-claim
+- as: cvp-common-hypershift
+  cron: '@yearly'
+  steps:
+    cluster_profile: aws-2
+    env:
```
Comment (nmars) [inline ci-operator/config/redhat-openshift-ecosystem/cvp/redhat-openshift-ecosystem-cvp-ocp-4.12.yaml:42]: Suggest adding a comment here: 
```
# For hypershift clusters, ICSPs must be specified in the HostedCluster spec.
```

File: ci-operator/step-registry/optional-operators/configure-brew-pull-secret-for-hypershift/optional-operators-configure-brew-pull-secret-for-hypershift-commands.sh [hunk 2]
```diff
@@ -2,17 +2,28 @@
...
-echo "The BREW_DOCKERCONFIGJSON variable is set"
+echo "BREW_DOCKERCONFIGJSON has been copied"
 
-echo "Copying the BREW_DOCKERCONFIGJSON variable to ${SHARED_DIR}/pull-secret-build-farm.json"
+echo "ICSP file is set to ${SHARED_DIR}/operators-image-content-sources.yaml"
 
-# copy brew pull secret to HostedCluster
-# >> appends
-echo "$BREW_DOCKERCONFIGJSON" >> ${SHARED_DIR}/pull-secret-build-farm.json
+cat > "${SHARED_DIR}/operators-image-content-sources.yaml" << EOF
```
Comment (nmars) [inline ci-operator/step-registry/optional-operators/configure-brew-pull-secret-for-hypershift/optional-operators-configure-brew-pull-secret-for-hypershift-commands.sh:14]: I suggest we put a comment here like:
```
# For hypershift clusters, ICSPs must be specified in the HostedCluster spec.
```

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Comment by gastoner** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-cvp-ocp-4.12-cvp-common-hypershift

**Pull request merged**